### DISCUSSION TITLE
Ability to Disable Suppression Lists

### DISCUSSION THREAD

**Discussion by leonvd6** [postalserver/postal#3231] [Feature suggestions]
Add the ability to disable suppression lists so that all messages get attempted and none held.

The only way I can see to achieve this at the moment is by running a CRON every minute to clear the **suppressions** table in the database.

**Comment by leonvd6** [comment]
Possibly we can also achieve it by running the following on a CRON:

`postal console
server = Server.find_by(name: 'your-postal-server-name')
server.message_db.messages(:where => {:held => 1, :scope => 'outgoing'}).each { |m| m.add_to_message_queue(:manual => true) }`

**Comment by willpower232** [comment]
Email addresses are only added to the suppression list after they fail 18 times by default so it is incredibly unlikely they are getting email at the minute due to a typo or massive technical issue.

Not sure why you wouldn't want this?

**Reply by leonvd6** [reply]
Hi Will,

Thank you for the quick reply! I was not aware of this. 

I see the following env can be used:`POSTAL_DEFAULT_SUPPRESSION_LIST_AUTOMATIC_REMOVAL_DAYS` to control how long an email should be supressed.

**Reply by leonvd6** [reply]
Just to clarify on our situation:

We have clients that send emails to addresses that don't exist yet. These emails bounce and then get suppressed. The following hours/days the user gets their email created but can't receive anything because the email has already been suppressed.

Unfortunately we cannot ask the client to not send to non-existent emails.

**Reply by willpower232** [reply]
Ah that is a bit unfortunate, you could certainly reduce the time they were on the suppression list but I would presume that if you set that setting to 0, all it does is restore them overnight.
